src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -2683,23 +2683,33 @@ private void getCallInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_RESO
                     (void*)ObjectToHandle(_compilation.NodeFactory.ReadyToRunHelper(ReadyToRunHelperId.ResolveGenericVirtualMethod, targetMethod));
                 pResult.nullInstanceCheck = false;
             }
-            else if ((flags & CORINFO_CALLINFO_FLAGS.CORINFO_CALLINFO_LDFTN) != 0)
+            else
```

Comment:
This changes the structure to fold together the virtual and non-virtual dispatch paths. While that is a simplification for now(since we only have R2R codegen), it is not in touch with longer term direction where we'll want to have more optimized codegen that handles the various cases in the most optimal form.